frontend/src/utils/api/goodShortVideos.ts

```diff
@@ -0,0 +1,37 @@
+import { initRecommendationsLanguages } from 'src/utils/recommendationsLanguages';
+import { PollsService } from 'src/services/openapi';
+import { Recommendation } from 'src/services/openapi';
+
+const shuffleRecommendations = (array: Recommendation[]) => {
+  // https://stackoverflow.com/a/12646864/188760
+  for (let i = array.length - 1; i > 0; i--) {
+    const j = Math.floor(Math.random() * (i + 1));
+    [array[i], array[j]] = [array[j], array[i]];
+  }
+};
+
+export const getGoodShortVideos = async (): Promise<Recommendation[]> => {
+  const languages = initRecommendationsLanguages();
```

Comment:
Done in 487c1b383b77e6b1c1b7ed1e718cf03faa62edfe.

I wonder if `usePreferredLanguages` should get the poll name itself. I think it's good to make it explicit that it depends on the poll, but adding a parameter for that is probably not the best solution. Maybe it should be in the name instead (`usePreferredLanguagesFromCurrentPoll` for example).

But it brings up another comment: `useCurrentPoll` is widely used but it does some calculations that are a bit wasteful to repeat. Its return value could be memoized with just the `poll` as a dependency, but we need to be sure that this value is not muted. Or we could have a `useCurrentPollName` which doesn't do any calculations, since `useCurrentPoll` is often used just for the name (25 times out of the 78 uses of `useCurrentPoll`).

All this is very minor and can be done later though.

Reply:
Thank you very much.

About `usePreferredLanguages` vs `usePreferredLanguagesFromCurrentPoll`, I don't have a strong preference. I think we can keep `usePreferredLanguages` for now, and refactor it if we optimize `useCurrentPoll`.

Regarding `useCurrentPoll` you are right, it could be good in theory to optimize it, but I'm not sure I understood correctly the risk of `poll` being muted, and the potential complexity added because of this. Can we talk about it on Discord or during our next meeting?